src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -786,16 +787,18 @@ private Result<UpdateSuccess, UpdateError> addTripToGraphAndBuffer(
     // Create StopTimes
     final List<StopTime> stopTimes = new ArrayList<>(stopTimeUpdates.size());
     for (int index = 0; index < stopTimeUpdates.size(); ++index) {
-      final StopTimeUpdate stopTimeUpdate = stopTimeUpdates.get(index);
+      final var added = new AddedStopTime(stopTimeUpdates.get(index));
       final var stop = stops.get(index);
 
       // Create stop time
       final StopTime stopTime = new StopTime();
       stopTime.setTrip(trip);
       stopTime.setStop(stop);
       // Set arrival time
-      if (stopTimeUpdate.hasArrival() && stopTimeUpdate.getArrival().hasTime()) {
-        final long arrivalTime = stopTimeUpdate.getArrival().getTime() - midnightSecondsSinceEpoch;
+      final var arrival = added.getArrivalTime();
+      if (arrival.isPresent()) {
+        final var delay = added.getArrivalDelay().orElse(0);
```

Comment:
Do we ever need to know if there was an arrival/departure value set? We seem to always use 0 as the fallback.

In that case the `orElse(0)` can also be encapsulated into `getArrivalDelay`. Can you check that?